File: examples-cs/examples-cs/examples-cs/example_03.cs
using EdgeJs;
using System;
using System.Threading.Tasks;

internal class example_03
    {
    internal static Task<Object> example_03_task(string firstName, string lastName)
    {
        string javascript = @"
            return function (input, callback) {
                try {
                    const message=`Hello ${input.firstName} ${input.lastName}`;
                    console.log(message);
                    callback(null, `result: ${message}`);
                } catch (error) {
                    callback(error, `result: ${message}`);
                }                
            }
        ";
        Func<Object, Task<Object>> func = Edge.Func(javascript);
        var input = new { firstName = firstName, lastName = lastName };
        return func(input);
    }

    internal static void exec()
    {
        string result = example_03_task("Ken", "Hill").Result as String;
        Console.WriteLine(result);

        //wait for keyboard
        Console.ReadLine();
    }
}

File: examples-node/example03.js
var edge = require('@agracio/edge-js/lib/edge');

var sayHello = edge.func('cs',function () {
/*
	// we must import standard libraries
	using System;
	using System.Threading.Tasks; 

	// edge class must be named Startup
	public class Startup 
		{
			// edge function must be async, named Invoke, single parameter of type dynamic
			public async Task<object> Invoke(dynamic input)
			{
				var hello=new Hello(input.firstName, input.lastName);
				return hello.ToString();
			}
		}

	// other classes can be defined as usual
	public class Hello
	{
		private string _firstName {get; set;}
		private string _lastName {get; set;}

		public Hello(string firstName, string lastName)
		{
			_firstName=firstName;
			_lastName=lastName;
		}

		public override string ToString() 
		{
			return String.Format("Hello {0} {1}", _firstName, _lastName); 
		}
	}	
*/
});

let firstName=process.argv[2];
let lastName=process.argv[3];

sayHello(
	{
		firstName: firstName,
		lastName: lastName
	}, 
	function (error, result) {
		if (error) throw error;
		console.log(result);
	}
);
